src/Microsoft.ML.Recommender/MatrixFactorizationStatic.cs

```diff
@@ -0,0 +1,125 @@
+        /// <param name="regularizationCoefficient">The frobenius norms of factor matrices.</param>
+        /// <param name="approximationRank">Rank of the two factor matrices whose product is used to approximate the consdered matrix</param>
+        /// <param name="learningRate">Initial learning rate.</param>
+        /// <param name="numIterations">Number of training iterations.</param>
+        /// <param name="advancedSettings">A delegate to set more settings.</param>
+        /// <param name="onFit">A delegate that is called every time the
+        /// <see cref="Estimator{TInShape, TOutShape, TTransformer}.Fit(DataView{TInShape})"/> method is called on the
+        /// <see cref="Estimator{TInShape, TOutShape, TTransformer}"/> instance created out of this. This delegate will receive
+        /// the model that was trained.  Note that this action cannot change the result in any way; it is only a way for the caller to
+        /// be informed about what was learnt.</param>
+        /// <returns>The predicted output.</returns>
+        public static Scalar<float> MatrixFactorization<T>(this RegressionContext.RegressionTrainers ctx,
+            Scalar<float> label, Key<T> matrixColumnIndex, Key<T> matrixRowIndex,
+            float regularizationCoefficient = 0.1f,
+            int approximationRank = 8,
+            float learningRate = 0.1f,
+            int numIterations = 20,
+            Action<MatrixFactorizationTrainer.Arguments> advancedSettings = null,
+            Action<MatrixFactorizationPredictor> onFit = null)
+        {
+            Contracts.CheckValue(label, nameof(label));
+            Contracts.CheckValue(matrixColumnIndex, nameof(matrixColumnIndex));
+            Contracts.CheckValue(matrixRowIndex, nameof(matrixRowIndex));
+
+            Contracts.CheckParam(regularizationCoefficient >= 0, nameof(regularizationCoefficient), "Must be non-negative");
+            Contracts.CheckParam(approximationRank > 0, nameof(approximationRank), "Must be positive");
+            Contracts.CheckParam(learningRate > 0, nameof(learningRate), "Must be positive");
+            Contracts.CheckParam(numIterations > 0, nameof(numIterations), "Must be positive");
+            Contracts.CheckValueOrNull(advancedSettings);
+            Contracts.CheckValueOrNull(onFit);
+
+            var rec = new MatrixFactorizationReconciler<T>((env, labelColName, matrixColumnIndexColName, matrixRowIndexColName) =>
+            {
+                var trainer = new MatrixFactorizationTrainer(env, labelColName, matrixColumnIndexColName, matrixRowIndexColName, advancedSettings:
+                    args =>
+                    {
+                        advancedSettings?.Invoke(args);
+                        args.Lambda = regularizationCoefficient;
+                        args.K = approximationRank;
+                        args.Eta = learningRate;
+                        args.NumIterations = numIterations;
+                    });
```

Comment:
@sfilipi is this pattern we agree to use in trainers?

Reply:
I think we resorted to use the advancedArgs to override the other params; so:

args.Lambda = regularizationCoefficient;
args.K = approximationRank;
args.Eta = learningRate;
 args.NumIterations = numIterations;

 advancedSettings?.Invoke(args);


---
In reply to: [229013738](https://github.com/dotnet/machinelearning/pull/1407#discussion_r229013738) [](ancestors = 229013738)

Reply:
Ok. Just swap the general settings and advanced settings.

---
In reply to: [229095838](https://github.com/dotnet/machinelearning/pull/1407#discussion_r229095838) [](ancestors = 229095838,229013738)